### DISCUSSION TITLE
Does Aspire Supports Running on Azure Service Fabric?

### DISCUSSION THREAD

**Discussion by xinlifoobar** [microsoft/aspire#2409] [Q&A] [answered]
Does Aspire Supports Running on Azure Service Fabric?

Even if there are feature overlaps like orchestration, service discovery and etc.. Aspire just like a modern replacement of Azure Service Fabric with attaction features including OTEL supports and uniform resource accesses. It would be helpful if there is a doc on how to add Aspire supports on Service fabric.. ideally, service fabric application could be modeled to keep Aspire hosts alive and the host keeps monitoring other apps.

**Comment by xinlifoobar** [comment]
Seems this is also related.. https://github.com/dotnet/aspire/issues/831

**Comment by xinlifoobar** [comment]
Hey... Any comments on this issue

**Comment by ReubenBond** [comment]
Aspire does not support deploying to Service Fabric today. Someone could create a Service Fabric provisioning tool, similar to the provisioning tool for Kubernetes linked in this thread: https://github.com/dotnet/aspire/issues/830.

**Comment by davidfowl** [comment, marked as answer]
Right! Service fabric is a runtime for both local development and a deployment target. Aspire's local orchestrator is for development and there's a process for turning that into a target deployment for another orchestrator. Since Aspire has a local experience, it's unclear if the service fabric emulator could be used in a compatible way with it as it exists today. As for deployment, a tool would need to be built to translate the manifest into a service fabric deployment, but that's out of scope for what we're targeting today.

**Reply by jennyf19** [reply]
@davidfowl is this still the case today? Or is it on the roadmap? Thanks!

**Reply by davidfowl** [reply]
It is still the case today and currently it is not on the roadmap.

**Reply by waynemunro** [reply]
> Right! Service fabric is a runtime for both local development and a deployment target. Aspire's local orchestrator is for development and there's a process for turning that into a target deployment for another orchestrator. Since Aspire has a local experience, it's unclear if the service fabric emulator could be used in a compatible way with it as it exists today. As for deployment, a tool would need to be built to translate the manifest into a service fabric deployment, but that's out of scope for what we're targeting today.

The local development is not emulated, it uses the same runtime.
